### PR TITLE
Start app deploy scripts

### PR DESCRIPTION
**Pull request by rogerhutchings**
This PR:

- Adds a Jenkins stage for deploying `app-*` folders as Docker images
- Prepares npm packages for publishing
- Adds a Dockerfile for `app-project`
- Removes `package-lock.json` files, as they were referring to outdated versions of local symlinked packages, and lerna doesn't support them yet (see lerna/lerna#913)
- Temporarily disables the test scripts for `react-components` and `panoptes-js` from running on Jenkins, as for some reason they aren't exiting properly. (cc #88 for `panoptes-js`)

cc #156

### CODE REVIEW

**Comment by rogerhutchings** [comment]
@adammcmaster would you mind setting up a deploy somewhere for https://hub.docker.com/r/zooniverse/fe-project/, just so we can see it during dev?

**Comment by adammcmaster** [comment]
@rogerhutchings Sure, I've added the necessary config for this to be deployed to the Swarm and to be accessed at https://fe-project.preview.zooniverse.org/. You'll need to do two things to get this working though:

1. The app needs to listen on port 80 when it's started in Docker.
2. Use the `hermes_wrapper.sh` script to deploy updates from the Jenkinsfile. [There's an example of what you'll need to do in the zoo-event-stats repo](https://github.com/zooniverse/zoo-event-stats/blob/master/Jenkinsfile#L21). Just replace all instances of `zoo-event-stats-staging` with `fe-project-stating`.

**Comment by rogerhutchings** [comment]
I've just made those changes @adammcmaster requested, so yes

File: Jenkinsfile [hunk 1]
```diff
@@ -1,23 +1,60 @@
 #!groovy
 
-node {
-  checkout scm
 
-  def dockerRepoName = 'zooniverse/front-end-monorepo'
-  def dockerImageName = "${dockerRepoName}:${env.BUILD_ID}"
-  def newImage = null
+timeout(20) {
```
Comment (srallen) [inline Jenkinsfile]: Do we need any handling in case of timeout? I suppose jenkins will just report a failure?

Comment (rogerhutchings) [reply Jenkinsfile]: Yeah, it'll just fail.

**Comment by rogerhutchings** [comment]
Cool - I've added those changes in @srallen

**Pull request merged**